### PR TITLE
fix junction/pnpm on window

### LINKED ISSUE #10469: On Windows, module resolver does not handle junctions or symlinks correctly
On Windows, Bun's module resolver does not detect correctly that a junction is a directory. We think it is a file. We try to read it from a file. And the code for reading files doesn't expect to receive a directory, causing it to return `Unexpected`.

Related issues:
- https://github.com/oven-sh/bun/issues/10013
- https://github.com/oven-sh/bun/discussions/10172
- #10125

### PR DESCRIPTION
**Pull request by paperclover**
### What does this PR do?

Fixes #10469

TODO: write tests

### CODE REVIEW

**Comment by Jarred-Sumner** [comment]
@paperdave you downgraded zig

File: src/sys.zig [hunk 1]
```diff
@@ -2096,23 +2096,128 @@ pub fn getMaxPipeSizeOnLinux() usize {
...
 
-pub fn existsOSPath(path: bun.OSPathSliceZ, file_only: bool) bool {
-    if (comptime Environment.isPosix) {
-        return system.access(path, 0) == 0;
+pub const WindowsFileAttributes = enum(windows.DWORD) {
+    invalid = windows.INVALID_FILE_ATTRIBUTES,
+    _,
+
+    pub fn isFile(this: WindowsFileAttributes) bool {
+        return @intFromEnum(this) & FILE_ATTRIBUTE_NORMAL == 0;
```
Comment (dylan-conway) [inline src/sys.zig]: is this supposed to be `!=`?

Comment (Jarred-Sumner) [reply src/sys.zig]: yes

Comment (Jarred-Sumner) [inline src/sys.zig]:
```suggestion
        return @intFromEnum(this) & FILE_ATTRIBUTE_NORMAL != 0;
```

**Pull request merged**